src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_sdfg.py

```diff
@@ -405,11 +407,7 @@ def visit_Lambda(
         # that is only the symbols that are used in the context of the lambda node
         self._add_sdfg_params(
             nsdfg,
-            [
-                gtir.Sym(id=p_name, type=p_type)
-                for p_name, p_type in lambda_symbols.items()
-                if p_name in used_symbols
-            ],
+            [gtir.Sym(id=p_name, type=p_type) for p_name, p_type in lambda_symbols.items()],
```

Comment:
I remember some discussions here, why did you change from the used to the lambda symbols.
For me the used symbols is smaller and thus more appropriate as it would not include the ones that are not used.

Reply:
Correct. I noticed that even filtering on `used_symbols` in combination with tuples in let-statements (coming PR) does not necessarily remove all redundant connectors, and the `PruneUnusedConnectors` transformation is anyway needed. So I thought I could just ignore the filtering here and let the `PruneUnusedConnectors` transformation do all job. And we agreed that it will be part of the `gt_simplify` function.